### PR TITLE
[added] table adds the horizontal scrolling functionality

### PR DESCRIPTION
**Pull request by little-snow-fox**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
为table组件添加了横向滚动功功能，并且滚动的时候表头跟随表体x轴进行滚动

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/components/table/nz-table.component.ts [hunk 1]
```diff
@@ -220,7 +225,7 @@ export class NzTableComponent implements AfterViewInit, OnInit {
   ngOnInit() {
     const scrollbarWidth = measureScrollbar();
```
Comment (wilsoncook) [inline src/components/table/nz-table.component.ts]: Unnecessarily? Includes `measureScrollbar` function refer to your modifications

File: src/components/table/nz-table.component.ts [hunk 2]
```diff
@@ -232,5 +237,13 @@ export class NzTableComponent implements AfterViewInit, OnInit {
   ngAfterViewInit() {
     this._isInit = true;
   }
+
+  scrollTable ($event) {
+    if (this.nzScroll != null &&  this.nzScroll.x != null) {
+      const body = <any>event.currentTarget;
```
Comment (wilsoncook) [inline src/components/table/nz-table.component.ts]: `event` do you mean `$event`? Pls checked by running these cases

File: src/components/table/nz-table.component.ts (hunk 1 above)
Comment (wilsoncook) [inline src/components/table/nz-table.component.ts]: This line includes `measureScrollbar` function are unnecessarily according to your modifications

Comment (little-snow-fox) [reply src/components/table/nz-table.component.ts]: Ok, thank

**Pull request closed without merging**